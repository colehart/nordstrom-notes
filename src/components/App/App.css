.App {
  height: 100vh;
  width: 100vw;
  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows: 160px 1fr;
}

.a-404 {
  width: 75%;
  margin: 0 auto;
  grid-row: 2 / 3;
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 40px 0;
  max-width: 555px;
}

.a-404 h2 {
  text-align: center;
}

.a-404 p {
  font-size: 1.15rem;
}

@media screen and (max-width: 890px) {
  .App {
    grid-template-rows: 225px 1fr;
  }
}