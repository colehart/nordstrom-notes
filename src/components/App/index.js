import React from 'react';
import { Route, Switch } from 'react-router-dom';
import Header from '../../containers/Header'
import NoteForm from '../../containers/NoteForm'
import NotesList from '../../containers/NotesList'
import './App.css';

export const App = () => {
  const noMatch = ({ location }) => (
    <div className='a-404'>
      <h2>404 - Path not found.</h2>
      <p>No routes match {location.pathname}.</p>
      <p>Please click the the 'Take Note' or 'Notable Notes' buttons or your browser's forward and back buttons to navigate to your desired destination. The Nordstrom Notes logo will return you to the Take Note page.</p>
    </div>
  )

  return (
    <div className="App">
      <Header />
      <Switch>
        <Route
          exact path='/'
          component={ NoteForm }
        />
        <Route
          exact path='/notes'
          component={ NotesList }
        />
        <Route
          component={ noMatch }
        />
      </Switch>
    </div>
  );
}

