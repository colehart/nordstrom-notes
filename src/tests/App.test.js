import React from 'react';
import { shallow } from 'enzyme';
import App from '../components/App';

describe('App', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App />)
  })

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
})