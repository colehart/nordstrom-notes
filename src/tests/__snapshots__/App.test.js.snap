// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches the snapshot 1`] = `
<div
  className="App"
>
  <Header />
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/notes"
    />
    <Route
      component={[Function]}
    />
  </Switch>
</div>
`;
